model/FieldType/DBComposite.php

```diff
@@ -223,11 +223,14 @@ public function hasField($field) {
 	 * @param string $field
 	 * @param mixed $value
 	 * @param bool $markChanged
+	 * @return $this
 	 */
 	public function setField($field, $value, $markChanged = true) {
+		$this->objCacheClear();
```

Comment:
Copy pasted from below:

The problem is we don't know what caches will be invalidated. E.g. if you modify URLSegment, then the Link, RelativeLink, etc caches are all invalid.

The use case is VERY limited, as rarely are objects modified during rendering.